### PR TITLE
Add BrowserSystemClock

### PR DESCRIPTION
**Pull request by BrendanAnnable**
Adds `BrowserSystemClock`, almost identical to `NodeSystemClock`, to be used in a browser context.

Also moves the common `Clock` interface and `FakeClock` into `shared/time`.

### CODE REVIEW

File: src/client/time/browser_clock.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { CancelTimer } from '../../shared/time/clock'
+import { Clock } from '../../shared/time/clock'
+
+function setTimeout(cb: (...args: any[]) => void, seconds: number): CancelTimer {
+  const handle = window.setTimeout(cb, seconds * 1e3)
+  return window.clearTimeout.bind(null, handle)
```
Comment (MonicaOlejniczak) [inline src/client/time/browser_clock.ts]: Can you change this to undefined? That is the preferred in typescript and is consistent with the rest of the codebase. The same for the one's below.

File: src/client/time/browser_clock.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { CancelTimer } from '../../shared/time/clock'
+import { Clock } from '../../shared/time/clock'
+
+function setTimeout(cb: (...args: any[]) => void, seconds: number): CancelTimer {
+  const handle = window.setTimeout(cb, seconds * 1e3)
```
Comment (MonicaOlejniczak) [inline src/client/time/browser_clock.ts]: Not a huge fan of the repeated constant. Either make it a variable or implement conversion functions.

Comment (BrendanAnnable) [inline src/client/time/browser_clock.ts]: Done!

File: src/client/time/browser_clock.ts (hunk 1 above)
Comment (BrendanAnnable) [inline src/client/time/browser_clock.ts]: Done!

**Pull request merged**